Branding

HerbertYiga opened the pull request:
# Pull Requests Requirements

- [ ] The PR title includes a brief description of the work done, including the
      Issue number if applicable.
- [ ] The PR includes a video showing the changes for the work done.
- [ ] The PR title follows conventional commit label standards.
- [ ] The changes confirm to the OpenElis Global x3
      [Styleguide and Design](https://uwdigi.atlassian.net/wiki/spaces/OG/pages/621346838/OpenELIS+Global+Style+Guide)
      documentation.
- [ ] The changes include tests or are validated by existing tests.
- [ ] I have read and agree to the Contributing
      [Guidelines](https://openelis-global.org/community/get-involved/) of this
      project.

## Summary

## Screenshots

[Add relevant screenshots here if applicable]

## Related Issue

[Add a link to the related issue or mention it here if applicable]

## Other

[Add any additional information or notes here]

Commit:
branding

Commit:
branding

mozzy11 commented:
Thanks @HerbertYiga .
Can you add a screen recording of the feature ??

HerbertYiga commented:
> Thanks @HerbertYiga . Can you add a screen recording of the feature ??

sure as soon as i am done with some more testing

Commit:
update

Commit:
update

mozzy11 commented:
@HerbertYiga can you fix the merge conflics and also run the formatters ?

Commit:
update

Commit:
update

CalebSLane commented on src/main/java/org/openelisglobal/sitebranding/valueholder/SiteBranding.java:
```diff
@@ -0,0 +1,142 @@
+    private static final long serialVersionUID = 1L;
+
+    @Id
+    @GeneratedValue(generator = "uuid-generator")
```
uuid is not used anywhere else in the codebase for OpenELIS db ids. Is there a specific reason to use it here?

mozzy11 replied:
By the way do we even need to persist this in the DB ?
we could  just keep it as  a frontend config  file , and just mounted on the frontend container.

ibacher replied:
@mozzy11 Because we want to let users dynamically change this in the system and it's more sane to do that in a multi-user system with a database and transactional control. In any case, we also allow users uploading files (for logos) and we need to have server-side validation that those files are correct, at which point, a database table is pretty small overhead.

mozzy11 commented:
@HerbertYiga why not just doing this entirely as a frontend config file instead of persisting it in the db ??
Creating an api for this looks like an overkill for me

ibacher committed:
Rebase on develop and adds a few TODOs:

1. Support for the Initializer feature with configuration in YAML
2. Extends various places to replace validation that string is a hex
   code. This enable us to support arbitrary, but valid CSS expressions
   including "blue", "hsl(128, 128, 54)", etc.
3. Removed the required privilege from the GET handler for branding since
   we need even unauthenticated user to leverage the branding availability

Note that this is just a progress commit. This PR still needs a bit more
ironing out before it's ready to merge.

ibacher committed:
Update 001-site-branding branch to match changes

ibacher committed:
Refactor and simplify implementation

This set of changes:

- Drops the idea of an "accent" color for now, as most of these elements
  are not style-able
- Switches to preferring a Carbon-native solution to branding
- Adds a new CSS variable for header color defaulting to OE's current header
- Removes the Branding component in favour of integrating branding into
  the core of the frontend
- Updates the Liquibase changeset to follow OE conventions

ibacher committed:
Fix tests and a few strings

ibacher committed:
Fix build issues

ibacher committed:
Fixes for tests, largely by making the branding locations configurable

ibacher committed:
More test fixes

ibacher committed:
Allow OE to return non-JSON objects via Spring

Also updates to using a slightly newer Docker API

ibacher committed:
Merge remote-tracking branch 'origin/develop' into 001-site-branding

ibacher committed:
Fixup

ibacher committed:
Make ETag effective

ibacher committed:
Use Spring's MediaType machinery

ibacher committed:
Some more headers for logos

ibacher committed:
Shift to using an integer for the db identifier

ibacher committed:
Only allow a single branding entry for now

mozzy11 committed:
Merge branch 'develop' into 001-site-branding

mozzy11 committed:
Merge branch 'develop' into 001-site-branding

mozzy11 committed:
fix e2e

mozzy11 committed:
fix site branding configuration

mozzy11 committed:
turn off cypress video

mozzy11 committed:
fix unit tests

mozzy11 committed:
fix display Notification

mozzy11 committed:
Merge branch 'develop' into 001-site-branding

mozzy11 committed:
fix formating

mozzy11 committed:
temoraly comment out failing tests

ibacher commented:
@mozzy11 Pretty sure this feature doesn't actually work yet...
